src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/Merge/ReferenceTokenCompositeSearchParamListRowGenerator.cs

```diff
@@ -0,0 +1,52 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage.TvpRowGeneration
+{
+    internal class ReferenceTokenCompositeSearchParamListRowGenerator : CompositeSearchParamRowGenerator<(ReferenceSearchValue component1, TokenSearchValue component2), ReferenceTokenCompositeSearchParamListRow>
+    {
+        private readonly ReferenceSearchParamListRowGenerator _referenceRowGenerator;
+        private readonly TokenSearchParamListRowGenerator _tokenRowGenerator;
+
+        public ReferenceTokenCompositeSearchParamListRowGenerator(
+            SqlServerFhirModel model,
+            ReferenceSearchParamListRowGenerator referenceRowGenerator,
+            TokenSearchParamListRowGenerator tokenRowGenerator,
+            SearchParameterToSearchValueTypeMap searchParameterTypeMap)
+            : base(model, searchParameterTypeMap)
+        {
+            _referenceRowGenerator = referenceRowGenerator;
+            _tokenRowGenerator = tokenRowGenerator;
+        }
+
+        internal override bool TryGenerateRow(short resourceTypeId, long resourceSurrogateId, short searchParamId, (ReferenceSearchValue component1, TokenSearchValue component2) searchValue, HashSet<ReferenceTokenCompositeSearchParamListRow> results, out ReferenceTokenCompositeSearchParamListRow row)
+        {
+            if (_referenceRowGenerator.TryGenerateRow(resourceTypeId, resourceSurrogateId, searchParamId, searchValue.component1, null, out var reference1Row) &&
```

Comment:
Is there a reason we are passing null to TryGenerateRow instead of passing "results"? 
Not sure if I'm missing it, but probably it'll would be required to avoid duplicates.

Reply:
If that's intentional, then we probably need a comment mentioning the reason why.

Reply:
I've noticed the same behavior in other places, like:

TokenStringCompositeSearchParamListRowGenerator 
TokenQuantityCompositeSearchParamListRowGenerator 
TokenNumberNumberCompositeSearchParamListRowGenerator 
TokenDateTimeCompositeSearchParamListRowGenerator

Reply:
It is intentional. I will put comment.